### DISCUSSION TITLE
Slow LAN Speed on nextcloud AIO

### DISCUSSION THREAD

**Discussion by H3rz3n** [nextcloud/all-in-one#4209] [Questions] [answered]
### Steps to reproduce
1. Install docker and nextcloud AIO
2. Upload a large file like an OS ISO
3. Local Upload Speed does not pass 15MB/s

### Expected behavior 
Local Upload Speed using full Gigabit speed
### Actual behavior 
Local upload speed barely pass 110Mbps

### Host OS 
The same problem is both on a machine with Debian 12 baremetal and on another machine with Proxmox and a VM of Fedora Server

#### Nextcloud AIO version 
Nextcloud Hub 7 28.0.2
#### Current channel 
Latest
#### Other valuable info 
The current network config is the following : OpenWRT Router connected to a raspberryPi Zero as with PiHole as DNS server. The two machine with Debian and Proxmox are connected with Cat6 cables. On the Debian machine I use Caddy as reverse proxy for both the nextcloud instance. I'm sorry for my bad english, I'm not a native speaker. Thank you for your time  :)

**Comment by H3rz3n** [comment]
### My current configuration.json

``` 
    "apache_port": "11000",
    "apache_ip_binding": "0.0.0.0",
    "talk_port": "3478",
    "nextcloud_upload_limit": "300G",
    "nextcloud_max_time": "360000",
    "nextcloud_memory_limit": "4096M",
    "nextcloud_enable_dri_device": "true",
    "domain": "cloud.my-nextcloud-domain.it",
    "borg_restore_password": "",
    "timezone": "Europe/Rome",
    "isClamavEnabled": 0,
    "isOnlyofficeEnabled": 0,
    "isCollaboraEnabled": 1,
    "isTalkEnabled": 0,
    "isTalkRecordingEnabled": 0,
    "isImaginaryEnabled": 1,
    "isFulltextsearchEnabled": 0,
    "isDockerSocketProxyEnabled": 1,
    "AIO_URL": "192.168.0.199:8080",
    "wasStartButtonClicked": 1,
    "install_latest_major": "",
    "collabora_dictionaries": "it_IT"
    
```

**Comment by szaimen** [comment]
Hi, so you have put the private ip-address of the server running aio as custom dns record into your pi-hole and all your devices are using the pi-hole as dns server?

**Reply by H3rz3n** [reply]
Yes, all the devices uses the pihole ad DNS server, I've checked it from their network panels. I tried to virtualize a Pihole on proxmox giving him more resource  but nothing changed. I also tried to us a VM with a new installation of Caddy as reverse proxy but nothing changed. What am i missing ? What else can I try ?

**Reply by szaimen** [reply]
Possibly one of your network infrastructure only have 100mbit connection? Also I hope you did test without wifi in the chain?

**Reply by H3rz3n** [reply]
Yes, the only possible cap is the raspberry pi zero that has a 100Mbps port but even creating a virtualized pihole in proxmox with a gigabit port it does not change anything. So I will exclude it because I have set the new DNS server on ther router and even manually on the interested devices. If I do an Iperf3 test on various devices in the lan network I can achieve full gigabit speed. Now I'm thinking to try Nginx as reverse proxy manager instead of Caddy.

**Comment by H3rz3n** [comment]
### Current network map :
![Schermata del 2024-02-13 19-40-26](https://github.com/nextcloud/all-in-one/assets/157974652/ab5dd04d-9d51-41bb-b4f0-0e604aa9b0e3)

**Reply by H3rz3n** [reply]
### Current Caddyfile on the Debian Server (NOT on PROXMOX) :

```
#Nextcloud
cloud.my-nextcloud-domain.it {                   
    header Strict-Transport-Security max-age=31536000;
    reverse_proxy 192.168.0.199:11000            #(Fedora VM)
}


#Manutenzione Nextcloud
aio.my-nextcloud-domain.it {
    reverse_proxy 192.168.0.199:8080 {           #(Fedora VM)
    transport http {
        tls_insecure_skip_verify
        }
    }
}
```

**Comment by szaimen** [comment]
Hi, since I cannot reproduce your issue on my test server (thr local upload speed is much higher), I still think this is caused by something in your network infrastructure so not a bug in AIO. Feel free to open a thread at https://help.nextcloud.com for community help on the topic.

**Reply by H3rz3n** [reply]
*UPDATE*

I tried to day to use Nginx instead of Caddy. Using the advised Nginx reverse proxy .conf I made a very small upgrade to 22MB/s (167Mbps) from 16MB/s (122Mbps). Still don't know where the issue is. I also tried to give more hardware resource to the software but nothing changes. I wish someone of you can give me an hint because I don't know what else try and this situation really slow down a lot my local lab use of Nextcloud.

**Reply by szaimen** [reply, marked as answer]
Hi, please create a thread on https://help.nextclooud.com thanks!

**Reply by szaimen** [reply]
Heere is another thing you could try: https://github.com/nextcloud/all-in-one/discussions/2475
